src/Resources/contao/dca/tl_article.php

```diff
@@ -2,15 +2,16 @@
 
 namespace Merconis\Core;
 
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+
 /*
  * Conditional Output
  */
-foreach ($GLOBALS['TL_DCA']['tl_article']['palettes'] as $paletteName => $palette)  {
-	if ($paletteName == '__selector__') {
-		continue;
-	}
-	$GLOBALS['TL_DCA']['tl_article']['palettes'][$paletteName] .= ';{lsShopConditionalOutput_legend},lsShopOutputCondition';
-}
+PaletteManipulator::create()
+    ->addLegend('lsShopConditionalOutput_legend')
+    ->addField('lsShopOutputCondition', 'lsShopConditionalOutput_legend')
+    ->applyToPalette('default', 'tl_article')
```

Comment:
Ich glaube, das macht nach dem Umschreiben nicht mehr dasselbe wie vorher. Vorher wurden alle Paletten in einer Schleife durchlaufen und die Ergänzung wurde in allen vorgenommen. Jetzt wird nur noch die default-Palette ergänzt, wenn ich das richtig sehe.

Es müsste vermutlich so gemacht werden wie in src/Resources/contao/dca/tl_content.php, Zeile 44 ff. (auch in diesem PR)

Reply:
Die DCA tl_article von Contao hat nur die default-Palette. Von dem her bräuchte man keine Schleife. Gab es denn einen konkreten Grund warum das im vorher so gemacht wurde?

Reply:
Die DCA tl_article von Contao hat nur die default-Palette. Von dem her bräuchte man keine Schleife. Gab es denn einen konkreten Grund warum das im vorher so gemacht wurde?

Reply:
Der Grund war dann sicher nur "Zukunftssicherheit". Die Annahme wäre, dass neu hinzukommende Paletten die von uns ergänzten Felder auf jeden Fall auch bräuchten. Die Umsetzung mit der Schleife würde das dann automatisch machen. Aber klar, solange es nur eine Palette gibt und wenn man davon ausgeht, dass man beim Hinzukommen einer weiteren Palette sowieso lieber nochmal drüber nachdenken möchte, ob diese Palette die Felder auch braucht, dann ist die Umsetzung im PR völlig i. O.